service-registry/src/main/java/org/apache/servicecomb/serviceregistry/ServiceRegistry.java

```diff
@@ -59,4 +59,38 @@ MicroserviceInstances findServiceInstances(String appId, String microserviceName
   Microservice getRemoteMicroservice(String microserviceId);
 
   Features getFeatures();
+
+  /**
+   * <p>
+   * Register a third party service if not registered before, and set it's instances into
+   * {@linkplain org.apache.servicecomb.serviceregistry.consumer.StaticMicroserviceVersions StaticMicroserviceVersions}.
+   * </p>
+   * <p>
+   * The registered third party service has the same {@code appId} and {@code environment} as this microservice instance has,
+   * and there is only one schema represented by {@code schemaIntfCls}, whose name is the same as {@code microserviceName}.
+   * </p>
+   * <em>
+   *   This method is for initializing 3rd party service endpoint config.
+   *   i.e. If this service has not been registered before, this service will be registered and the instances will be set;
+   *   otherwise, NOTHING will happen.
+   * </em>
+   *
+   * @param microserviceName name of the 3rd party service, and this param also specifies the schemaId
+   * @param version version of this 3rd party service
+   * @param schemaIntfCls the producer interface of the service. This interface is used to generate swagger schema and
+   * can also be used for the proxy interface of RPC style invocation.
+   * @param instances the instances of this 3rd party service. Users only need to specify the endpoint information, other
+   * necessary information will be generate and set in the implementation of this method.
+   */
+  void registerMicroserviceMapping(String microserviceName, String version, Class<?> schemaIntfCls,
```

Comment:
It's better to put schemaIntfCls as last parameter, since we can extend it to schemaIntfCls... to support several schemas. I think this is quite normal in many scenarios like register a gateway service as a third party service.

Reply:
Fixed.